### PR TITLE
[PIPE-325] raw.broker_subaward on delta

### PR DESCRIPTION
**Pull request by dpb-bah**
**Description:**
Updating TABLE_SPEC and added Broker Subaward delta model such that `load_table_to_delta --destination-table broker_subaward` copies the `subaward` table from the Broker to `raw.broker_subaward` on delta.

**Technical details:**
* Includes a `get_broker_jdbc_url()` function to be used if the TABLE_SPEC `broker` is set to True
* Added support for config values regarding `DATA_BROKER_DATABASE_URL` along with test updates

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2.  API documentation updated
3. [x] Necessary PR reviewers:
    - [x] Backend
    - Frontend <OPTIONAL>
    - Operations <OPTIONAL>
    - Domain Expert <OPTIONAL>
4. Matview impact assessment completed
5. Frontend impact assessment completed
6. [x] Data validation completed
7. Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-123](https://federal-spending-transparency.atlassian.net/browse/PIPE-235):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download) (takes about 20 minutes to load)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/config/envs/default.py [hunk 1]
```diff
@@ -75,125 +81,149 @@ def __new__(cls, *args, **kwargs):
...
+                    user=values[f"{db_conf_prefix}_USER"],
+                    password=values[f"{db_conf_prefix}_PASSWORD"].get_secret_value(),
+                    host=values[f"{db_conf_prefix}_HOST"],
+                    port=values[f"{db_conf_prefix}_PORT"],
+                    path="/" + values[f"{db_conf_prefix}_NAME"] if values[f"{db_conf_prefix}_NAME"] else None,
+                )
+                values = eval_default_factory_from_root_validator(cls, values, db_url_conf_name, lambda: str(pg_dsn))
+
+        # the broker config values are less required than the usas db values. if available, check for consistency.
+        if values[db_url_conf_name]:
```
Comment (sethstoudenmier) [inline usaspending_api/config/envs/default.py:157]: Based on the comment you have here should this be `values.get()` instead of `values[]`?

**Review by sethstoudenmier — DISMISSED** [review]
Approving since my comments are more about suggested changes for convention. Overall LGTM.

File: .env.template [hunk 2]
```diff
@@ -35,10 +35,10 @@ USASPENDING_DB_PASSWORD=usaspender
 
 # Configuration values for a connection string to a Broker database
 #    Only necessary for some management commands
-BROKER_USER=root
```
[the remark this answers is not in the document]
Comment (dpb-bah) [reply .env.template:38]: Yes these values are based on going through the broker docker setup that includes a broker DB.

File: usaspending_api/config/envs/default.py (hunk 1 above)
Comment (dpb-bah) [reply usaspending_api/config/envs/default.py:157]: I can change it to the .get() equivalent. I believe usually this field is available, just defaults to `None`, but couldn't hurt to be more careful.

**Review by sethstoudenmier — APPROVED** [review]
LGTM

**Pull request merged**